pkg/ccl/importccl/import_stmt.go

```diff
@@ -685,11 +690,45 @@ func importPlanHook(
 		if err != nil {
 			return err
 		}
-		return <-errCh
+		return copyToResultsChannel(ctx, errCh, resultsCh, jobResultsCh)
 	}
 	return fn, backupccl.RestoreHeader, nil, false, nil
 }
 
+// copyToResultsChannel exists to deal with the fact that the resultsCh hooked
+// up to the conn executor does not notify the plan hook when the connection
+// goes away. When that connection goes away however, the context will be
+// canceled. Because the started job needs to send on the resultsFromJobs
+// channel in order to complete, this function launches a goroutine to consume
+// from that channel in the case of context cancellation.
+func copyToResultsChannel(
```

Comment:
I think this helper should live in the `jobs` package ? assuming everyone using resultsCh has the same concern.